libs/common/src/auth/services/token.service.ts

```diff
@@ -599,20 +618,26 @@ export class TokenService implements TokenServiceAbstraction {
           // Fall back to disk storage for refresh token
           decryptedRefreshToken = await this.singleUserStateProvider
             .get(userId, REFRESH_TOKEN_DISK)
-            .update((_) => refreshToken);
+            .update(() => refreshToken, {
+              shouldUpdate: (previousValue) => previousValue !== refreshToken,
+            });
         }
 
         return decryptedRefreshToken;
       }
       case TokenStorageLocation.Disk:
         return await this.singleUserStateProvider
           .get(userId, REFRESH_TOKEN_DISK)
-          .update((_) => refreshToken);
+          .update(() => refreshToken, {
+            shouldUpdate: (previousValue) => previousValue !== refreshToken,
+          });
 
       case TokenStorageLocation.Memory:
         return await this.singleUserStateProvider
           .get(userId, REFRESH_TOKEN_MEMORY)
-          .update((_) => refreshToken);
+          .update(() => refreshToken, {
```

Comment:
```suggestion
          .update((_) => refreshToken);
```
Not my ownership, but the linked epic describing the change clearly states that this applies to disk I/O only. Subsequently, there is no reason to apply it to memory state.